src/mbed_tools/build/build.py

```diff
@@ -22,15 +22,18 @@ def build_project(build_dir: pathlib.Path, target: Optional[str] = None) -> None
     _cmake_wrapper("--build", str(build_dir), *target_flag)
 
 
-def generate_build_system(source_dir: pathlib.Path, build_dir: pathlib.Path, profile: str) -> None:
+def generate_build_system(
+    source_dir: pathlib.Path, build_dir: pathlib.Path, profile: str, generator: str = "Ninja"
+) -> None:
     """Configure a project using CMake.
 
     Args:
         source_dir: Path to the CMake source tree.
         build_dir: Path to the CMake build tree.
         profile: The Mbed build profile (develop, debug or release).
+        generator: The CMake build system generator to use.
     """
-    _cmake_wrapper("-S", str(source_dir), "-B", str(build_dir), "-GNinja", f"-DCMAKE_BUILD_TYPE={profile}")
+    _cmake_wrapper("-S", str(source_dir), "-B", str(build_dir), f"-G{generator}", f"-DCMAKE_BUILD_TYPE={profile}")
```

Comment:
What's the motivation to add this flag? Why not just drop down into CMake to do extra non-default stuff? Add this in the commit message if the justification is strong.